inst/bin/lintr

```diff
@@ -1,20 +1,37 @@
 #!/usr/bin/env Rscript
-files <- commandArgs(trailing = TRUE)
+
+
+"Run lintr on R files during a precommit.
+Usage:
+  lintr [--warn_only] <files>...
+Options:
+  --warn_only  Print lint warnings instead of blocking the commit. Should be
+               used with `verbose: True` in `.pre-commit-config.yaml`.
+               Otherwise, lints will never be shown to the user.
+" -> doc
+
+arguments <- docopt::docopt(doc)
 
 if (git2r::in_repository()) {
   git_status <- git2r::status(staged = FALSE)
   if (any(unlist(git_status) == ".lintr")) {
-    stop("Unstaged changes to .lintr file. Stage the .lintr file or discard the changes to it. ", call. = FALSE)
+    stop(
+      "Unstaged changes to .lintr file. Stage the .lintr file or discard ",
+      "the changes to it. ",
+      call. = FALSE
+    )
   }
 }
 
-out <- lapply(files, function(path) {
+for (path in arguments$files) {
   lints <- lintr::lint(path)
   if (length(lints) > 0) {
-    stop(
-      "File ", path, " is not lint free\n",
-      "Run lintr::lint('", path, "') to check the lintr output.",
-      call. = FALSE
-    )
+    cat("File `", path, "` is not lint free\n", sep = "")
+    rendered_lints <- capture.output(print(lints))
+    cat(rendered_lints, sep = "\n")
   }
-})
+
+  if (!arguments$warn_only) {
+    stop("File ", path, " is not lint free", call. = FALSE)
```

Comment:
I believe you want this if condition within `if (length(lints) > 0)`? Otherwise, the hook will fail also if you have zero lints but did not set the `--warn-only` flag. Right?

Reply:
I believe this is the reason why [one test](https://travis-ci.org/lorenzwalthert/precommit/builds/621338796#L1014) fails currently.

Reply:
Totally right!